src/axom/core/Map.hpp

```diff
@@ -0,0 +1,487 @@
+          m_list[m_head].key = key;
+          m_list[m_head].value = value;
+          m_size++;
+        }
+        else{
+          while(m_list[ind].next != -1){
+            if(m_list[ind].key == key){           
+              axom_map::Pair<Key, T> ret(&(m_list[ind]), false);
+              return ret;
+            }
+            ind = m_list[ind].next;
+          }
+          m_list[ind].next = m_free;
+          ind = m_free;
+          m_free = m_list[m_free].next;
+          m_list[ind].next = -1;
+          m_list[ind].key = key;
+          m_list[ind].value = value;
+          m_size++;
+        }
+        axom_map::Pair<Key, T> ret(&(m_list[ind]), true);
+        return ret;
+      }
+      axom_map::Pair<Key, T> ret(&(m_end), false);
+      return ret;
+    } 
+  
+    /*!
+     * \brief Removes item with the given key from the linked list, if item exists. Relinks accordingly.
+     *
+     * \param [in] key the key of the item to be found and removed.
+     *
+     * \return Returns true if item was found and removed, false otherwise. 
+     *  
+     */ 
+    bool remove(Key key){
+     if(m_head == -1){
+       return false;
+     }   
+     IndexType ind = m_head;
+     IndexType prev = -1;
+     do{
+       if(m_list[ind].key == key){
+         if(prev != -1){
+           m_list[prev].next = m_list[ind].next;
+         }
+         if(ind == m_head){m_head = m_list[ind].next;}
+         m_list[ind].next = m_free;
+         m_free = ind;
+         m_size--;
+         return true;
+       }
+       prev = ind;
+       ind = m_list[ind].next;
+     } while(ind != -1);
+     return false;
+   }
+
+   
+   /*!
+    * \brief Returns pointer to node containing key-value pair matching provided key, if possible.
+    *
+    * \param [in] key the key of the item to be searched for.
+    *
+    * \return Returns pointer to the requested node if possible, and to special end node otherwise.
+    *
+    */    
+   axom_map::Node<Key, T>& find(Key key){
+     if(m_head == -1){
+       return m_end;
+     }
+     IndexType ind = m_head;
+     do{
+       if(m_list[ind].key == key){
+         return m_list[ind];
+       }
+
+       ind = m_list[ind].next;
+    }while(ind != -1);
+    return m_end;
+   }
+    
+  /*!
+   * \brief Returns maximum number of items that can be stored in this Bucket instance.
+   *
+   * \return Returns maximum number of items that can be stored in this Bucket instance.
+   *
+   */   
+  int get_capacity(){ return m_capacity; }
+
+  /*!
+   * \brief Returns current number of items in this Bucket instance.
+   *
+   * \return Returns current number of items in this Bucket instance.
+   */ 
+  int get_size(){ return m_size; }   
+  ///@}
+  
+
+    axom_map::Node<Key, T> * m_list;   
+    axom_map::Node<Key, T> m_end;
+    IndexType m_head;
+    IndexType m_free;
+    int m_capacity, m_size;
+  
+  }; 
+
+  /// @}
+}
+
+/*!
+ * \class Map
+ *
+ * \brief Provides a hashmap implementation, relying upon chaining.
```

Comment:
"Relying upon chaining" to do what?  As a computer scientist, I think you mean to deal with hash collisions; please state so.